rclcpp/include/rclcpp/rate.hpp

```diff
@@ -42,19 +45,20 @@ using std::chrono::duration_cast;
 using std::chrono::nanoseconds;
 
 template<class Clock = std::chrono::high_resolution_clock>
-class GenericRate : public RateBase
+class [[deprecated("use rclcpp::Rate class instead of GenericRate")]] GenericRate
```

Comment:
Shouldn't this derive from RateBase again?

Reply:
Since of meaning of clock types has been updated (ROS-respecitve clock type added), new `rcl_clock_type_t get_type()` method [was introduced](https://github.com/ros2/rclcpp/pull/2123/files#diff-c6ecfbeee7b7384c9e72a8984cc2fc718415b3bb741c0ea41ffc1d9b0718fea7R39) in a `RateBase` basic class as a alternative/replacement for `bool is_steady()` one (I would like to completely remove `is_steady()` method from on now as unnecessary, but since we need to support old API for compativility for `rclcpp::Rate` and `rclcpp::WallRate` classes, `is_steady` was remain). `GenericRate` if being a child-class of `RateBase` now should implement `get_type()` method. But since `GenericRate` - is deprecated class, it might be overkill, I believe.

Therefore `GenericRate` was taken out of parenthood of `RateBase` remaining the same functionality as it previously has; while new `RateBase -> Rate -> WallRate/ROSRate` lineage was introduced instead.

Reply:
Sorry to be late to get back to you. What we need to deprecate is `RateBase::is_steady` since the application can get the clock type?

Reply:
The problem with removing `GenericRate` from the hierarchy is that this breaks certain applications.  Consider for example if a downstream node has `std::vector<RateBase>` in their code.  In that case, they will no longer be able to store a `GenericRate` in there, since it is no longer a child class of `RateBase`.

So I think we should add this back to the hierarchy (i.e. make this a child of `RateBase`), and implement `get_type()` as returning either `RCL_SYSTEM_TIME` (if `Clock::is_steady` is false), or `RCL_STEADY_TIME` (if `Clock::is_steady` is true).

Reply:
@fujitatomoya, OK. Thanks for reaching me. I've made `is_steady()` method to be deprecated for all classes in [this](https://github.com/ros2/rclcpp/pull/2123/commits/5f08e09bb048558689fb4e54e28b94bd1f9340ca) commit.

@clalancette, Got your point. This is indeed makes sense, so `RateBase` -> `GenericRate` relationship was restored in the [same](https://github.com/ros2/rclcpp/pull/2123/commits/5f08e09bb048558689fb4e54e28b94bd1f9340ca) commit.